src/applications/claims-status/components/ClaimCard/ClaimCardLink.jsx

```diff
@@ -1,23 +1,43 @@
 import React from 'react';
-import { Link } from 'react-router-dom-v5-compat';
+import { useNavigate } from 'react-router-dom-v5-compat';
+import { VaLink } from '@department-of-veterans-affairs/component-library/dist/react-bindings';
 import PropTypes from 'prop-types';
 
+/**
+ * We use e.preventDefault() + navigate() instead of relying on href alone
+ * because VaLink with href causes a full browser navigation, which:
+ * - Is slower (reloads entire page)
+ * - Loses React state
+ *
+ * This is how 'to=""' worked in the previous version of this component.
+ */
 export default function ClaimCardLink({
   ariaLabel,
   href,
   text = 'Details',
   onClick,
 }) {
+  const navigate = useNavigate();
+
+  const handleClick = e => {
+    // Prevent browser's default navigation (full page reload)
+    e.preventDefault();
+    if (onClick) {
+      onClick(e);
+    }
+    // Use React Router for client-side navigation
+    navigate(href);
+  };
+
```

Comment:
This update had to be made to keep the same behavior as the `Link` component from react.